enterprise/admin/views.py

```diff
@@ -229,28 +232,47 @@ def _enroll_users(cls, enterprise_customer, emails, course_id, mode, request):
             course_id: The ID of the course in which we want to enroll
             mode: The enrollment mode the users will be enrolled in the course with
             request: The HTTP request the enrollment is being created by
+            notify: Whether to notify (by email) the users that have been enrolled
         """
         enrolled = []
         non_existing = []
         failed = []
         enrollment_client = EnrollmentApiClient()
+        course_details = CourseApiClient().get_course_details(course_id)
+        reverse_method = get_reverse_by_site(request, enterprise_customer.site)
+        course_url = reverse_method('about_course', args=(course_id,))
+        email_conn = mail.get_connection()
+        email_conn.open()
         for email in emails:
             try:
-                username = User.objects.get(email=email).username
+                user = User.objects.get(email=email)
             except User.DoesNotExist:
                 non_existing.append(email)
                 continue
             try:
-                enrollment_client.enroll_user_in_course(username, course_id, mode)
+                enrollment_client.enroll_user_in_course(user.username, course_id, mode)
             except HttpClientError as exc:
                 failed.append(email)
                 error_message = json.loads(exc.content.decode()).get("message", "No error message provided.")
                 logging.error(
                     "Error while enrolling user %(user)s: %(message)s",
-                    dict(user=username, message=error_message),
+                    dict(user=user.username, message=error_message),
                 )
             else:
                 enrolled.append(email)
+                if notify:
+                    send_notification_message(
+                        user=user,
+                        enrolled_in={
+                            'name': course_details['name'],
+                            'url': course_url,
+                            'type': 'course',
+                            'start': datetime.datetime.strptime(course_details['start'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
The EdX API client is pretty sparse; it doesn't actually know much about the API structure. And, since JSON doesn't actually have a standard datetime format or way to specify that a field is, in fact, a datetime, there's not a reliable way to automatically parse a datetime from JSON.